<launch>
    <arg name="topic_from_controller" value="/center_pid/control_effort"/>
    <arg name="topic_from_plant" value="/center_pid/state"/>
    <arg name="setpoint_topic" value="/center_pid/setpoint"/>
    <node name="drag_centerline_planner" pkg="rr_iarrc" type="drag_centerline_planner" output="screen">
        <param name="speed" type="double" value="9.0" /> <!-- 3.1-->

        <param name="useHistogramFinder" type="bool" value="false" />

        <param name="maxTurnLimitRadians" type="double" value="0.16" />

        <param name="subscription_node" type="string" value="/camera_center/lines/detection_img_transformed"/>

        <param name="control_topic" value="$(arg topic_from_controller)" />
        <param name="state_topic" value="$(arg topic_from_plant)" />
        <param name="setpoint_topic" value="$(arg setpoint_topic)" />
    </node>

    <node name="drag_centerline_planner_pid" pkg="pid" type="controller">
        <param name="Kp" value="0.002" />
        <param name="Ki" value="0.0" />
        <param name="Kd" value="0.0005" />

        <param name="topic_from_controller" value="$(arg topic_from_controller)" />
        <param name="topic_from_plant" value="$(arg topic_from_plant)" />
        <param name="setpoint_topic" value="$(arg setpoint_topic)" />
    </node>
</launch>
